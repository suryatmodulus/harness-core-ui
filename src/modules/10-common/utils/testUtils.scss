/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
