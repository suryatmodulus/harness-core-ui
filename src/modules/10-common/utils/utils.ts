/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
