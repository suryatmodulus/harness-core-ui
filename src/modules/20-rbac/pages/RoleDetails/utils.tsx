/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
