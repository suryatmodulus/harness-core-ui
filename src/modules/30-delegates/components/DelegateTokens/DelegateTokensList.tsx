/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useMemo, useState } from 'react'
import { TableV2, Button, Layout, Popover, ButtonVariation, Icon } from '@wings-software/uicore'
import type { CellProps, Renderer, Column } from 'react-table'
import ReactTimeago from 'react-timeago'
import { Menu, MenuItem, Classes, Position } from '@blueprintjs/core'
import { useStrings } from 'framework/strings'
import type { DelegateTokenDetails } from 'services/portal'

import { useRevokeTokenModal } from './modals/useRevokeTokenModal'
import { useMoreTokenInfoModalModal } from './modals/useMoreTokenInfoModal'

import css from './DelegateTokens.module.scss'

interface TokensListViewProps {
  onRevokeSuccess?: () => void
  delegateTokens: DelegateTokenDetails[]
  reload?: () => Promise<void>
  gotoPage?: any
}

type CustomColumn<T extends Record<string, any>> = Column<T> & {
  reload?: () => Promise<void>
}

const DelegateTokensList: React.FC<TokensListViewProps> = props => {
  const { delegateTokens, reload, onRevokeSuccess } = props
  const { getString } = useStrings()

  const { openRevokeTokenModal } = useRevokeTokenModal({ onSuccess: onRevokeSuccess })
  const { openMoreTokenInfoModal } = useMoreTokenInfoModalModal({})

  const RenderColumnName: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => (
    <span className={`${css.tokenNameColumn} ${css.tokenCellText}`}>
      <Icon name="key" size={28} margin={{ right: 'small' }} />
      {row.original.name}
    </span>
  )
  const RenderColumnCreatedAt: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => (
    <span className={css.tokenCellText}>
      {row.original.createdAt && <ReactTimeago date={row.original.createdAt} />}
    </span>
  )
  const RenderColumnCreatedBy: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => (
    <span className={css.tokenCellText}>
      {row.original.createdBy
        ? row.original.createdBy?.name?.toLowerCase()
        : getString('delegates.tokens.createdBySystem')}
    </span>
  )
  const RenderColumnStatus: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => (
    <span className={css.tokenCellText}>
      {row.original.status === 'ACTIVE' ? getString('active') : getString('delegates.tokens.revoked')}
    </span>
  )
  const RenderColumnActions: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => (
    <span className={css.tokenCellText}>
      {row.original.status !== 'REVOKED' && (
        <Button
          variation={ButtonVariation.SECONDARY}
          onClick={e => {
            e.stopPropagation()
            openRevokeTokenModal(row.original.name || '')
          }}
        >
          {getString('delegates.tokens.revoke')}
        </Button>
      )}
    </span>
  )
  const RenderColumnMenu: Renderer<CellProps<DelegateTokenDetails>> = ({ row }) => {
    const [menuOpen, setMenuOpen] = useState(false)
    return (
      <Layout.Horizontal className={css.menuColumn}>
        <Popover
          isOpen={menuOpen}
          onInteraction={nextOpenState => {
            setMenuOpen(nextOpenState)
          }}
          className={Classes.DARK}
          position={Position.RIGHT_TOP}
        >
          <Button
            minimal
            icon="Options"
            onClick={e => {
              e.stopPropagation()
              setMenuOpen(true)
            }}
          />
          <Menu style={{ minWidth: 'unset' }}>
            <MenuItem
              icon="edit"
              text={getString('delegates.tokens.moreInfo')}
              onClick={() => openMoreTokenInfoModal(row.original.name || '')}
            />
          </Menu>
        </Popover>
      </Layout.Horizontal>
    )
  }

  const columns: CustomColumn<DelegateTokenDetails>[] = useMemo(
    () => [
      {
        Header: getString('name').toUpperCase(),
        accessor: 'name',
        id: 'name',
        width: '33%',
        Cell: RenderColumnName
      },
      {
        Header: getString('createdAt').toUpperCase(),
        accessor: 'createdAt',
        id: 'createdAt',
        width: '20%',
        Cell: RenderColumnCreatedAt
      },
      {
        Header: getString('createdBy').toUpperCase(),
        accessor: 'createdBy',
        id: 'createdBy',
        width: '20%',
        Cell: RenderColumnCreatedBy
      },
      {
        Header: getString('status').toUpperCase(),
        accessor: 'status',
        id: 'activity',
        width: '15%',
        Cell: RenderColumnStatus
      },
      {
        Header: '',
        width: '10%',
        id: 'actions',
        Cell: RenderColumnActions,
        reload: reload,
        disableSortBy: true
      },
      {
        Header: '',
        width: '3%',
        id: 'menu',
        Cell: RenderColumnMenu,
        reload: reload,
        disableSortBy: true
      }
    ],
    []
  )

  return (
    <TableV2<DelegateTokenDetails>
      sortable={true}
      className={css.table}
      columns={columns}
      data={delegateTokens}
      name="TokensListView"
      onRowClick={({ name }) => {
        openMoreTokenInfoModal(name || '')
      }}
    />
  )
}

export default DelegateTokensList
