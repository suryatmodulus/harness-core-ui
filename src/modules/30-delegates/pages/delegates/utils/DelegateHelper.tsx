/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
