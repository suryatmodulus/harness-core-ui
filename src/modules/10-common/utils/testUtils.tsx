/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { queryByAttribute } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uicore'
import qs from 'qs'

import { enableMapSet } from 'immer'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?(key: string): string
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultFeaturesValues,
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <StringsContext.Provider value={{ data: stringsData as any, getString }}>
      <AppStoreContext.Provider
        value={{
          featureFlags: {
            FEATURE_ENFORCEMENT_ENABLED: true
          },
          updateAppStore: () => void 0,
          currentUserInfo: {},
          ...defaultAppStoreValues
        }}
      >
        <LicenseStoreContext.Provider
          value={{
            versionMap: {},
            licenseInformation: {},
            CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            updateLicenseStore: () => void 0,
            ...defaultLicenseStoreValues
          }}
        >
          <PermissionsContext.Provider
            value={{
              permissions: new Map<string, boolean>(),
              requestPermission: () => void 0,
              checkPermission: () => true,
              cancelRequest: () => void 0,
              ...defaultPermissionValues
            }}
          >
            <FeaturesContext.Provider
              value={{
                features: new Map<FeatureIdentifier, FeatureDetail>(),
                featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                getEdition: () => {
                  return undefined
                },
                requestFeatures: () => void 0,
                requestLimitFeature: () => void 0,
                checkFeature: () => {
                  return defaultReturn
                },
                checkLimitFeature: () => {
                  return defaultReturn
                },
                getRestrictionType: () => {
                  return undefined
                },
                ...defaultFeaturesValues
              }}
            >
              <Router history={history}>
                <ModalProvider>
                  <RestfulProvider base="/">
                    <BrowserView enable={props.enableBrowserView}>
                      <Switch>
                        <Route exact path={path}>
                          {props.children}
                        </Route>
                        <Route>
                          <CurrentLocation />
                        </Route>
                      </Switch>
                    </BrowserView>
                  </RestfulProvider>
                </ModalProvider>
              </Router>
            </FeaturesContext.Provider>
          </PermissionsContext.Provider>
        </LicenseStoreContext.Provider>
      </AppStoreContext.Provider>
    </StringsContext.Provider>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)
