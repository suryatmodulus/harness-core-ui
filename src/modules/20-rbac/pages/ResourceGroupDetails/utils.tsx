import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
