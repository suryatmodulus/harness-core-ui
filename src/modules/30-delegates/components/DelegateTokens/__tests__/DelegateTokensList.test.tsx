/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import type { DelegateTokenDetails } from 'services/portal'
import { TestWrapper } from '@common/utils/testUtils'
import DelegateTokensList from '../DelegateTokensList'

const mockTokens: DelegateTokenDetails[] = [
  {
    accountId: 'dsaddefrfr',
    createdAt: 4342354343553,
    createdBy: { name: 'Admin' },
    identifier: 'dsfevrwv',
    name: 'Token 1',
    status: 'ACTIVE',
    uuid: 'debterbteb',
    value: 'fjon2kjh3rl4'
  }
]

jest.mock('@common/exports', () => ({}))

jest.mock('services/portal', () => ({}))

jest.mock('services/cd-ng', () => ({}))

describe('Delegate Tokens list', () => {
  test('render delegate token list', async () => {
    const { container } = render(
      <TestWrapper>
        <DelegateTokensList delegateTokens={mockTokens} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
