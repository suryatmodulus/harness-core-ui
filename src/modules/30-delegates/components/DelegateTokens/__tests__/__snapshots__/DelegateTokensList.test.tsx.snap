// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delegate Tokens list render delegate token list 1`] = `
<div>
  <div
    class="TableV2--table table"
  >
    <div
      class="TableV2--header"
      role="row"
    >
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 33%;"
        title="Toggle SortBy"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewname"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            NAME
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 20%;"
        title="Toggle SortBy"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewcreatedAt"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            CREATEDAT
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 20%;"
        title="Toggle SortBy"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewcreatedBy"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            CREATEDBY
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 15%;"
        title="Toggle SortBy"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewactivity"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            STATUS
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height="15"
            viewBox="0 0 16 16"
            width="15"
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 10%;"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewactions"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            
          </p>
        </div>
      </div>
      <div
        class="TableV2--cell TableV2--sortable"
        colspan="1"
        role="columnheader"
        style="width: 3%;"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="TokensListViewmenu"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            
          </p>
        </div>
      </div>
    </div>
    <div
      class="TableV2--body"
    >
      <div
        class="TableV2--row TableV2--card TableV2--clickable"
        role="row"
      >
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 33%;"
        >
          <span
            class="tokenNameColumn tokenCellText"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--margin-right-small"
              data-icon="key"
            >
              <svg
                height="28"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  opacity="0.6"
                >
                  <circle
                    cx="14"
                    cy="14"
                    fill="#D0E2FF"
                    r="14"
                  />
                  <path
                    d="M16.488 6.533a4.977 4.977 0 00-4.977 4.978c0 .445.064.874.174 1.285l-5.152 5.151v3.519h3.52l5.151-5.151c.41.11.84.174 1.285.174a4.978 4.978 0 000-9.956zm-6.95 13.69l-1.76-.002v-1.759l4.426-4.426a4.997 4.997 0 001.76 1.76l-4.426 4.426zm6.95-4.979a3.733 3.733 0 110-7.466 3.732 3.732 0 013.734 3.733 3.732 3.732 0 01-3.734 3.733zm2.125-4.613a1.243 1.243 0 11-2.489 0 1.245 1.245 0 112.489 0z"
                    fill="#0042A9"
                    fill-rule="nonzero"
                  />
                </g>
              </svg>
            </span>
            Token 1
          </span>
        </div>
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 20%;"
        >
          <span
            class="tokenCellText"
          >
            dummy date
          </span>
        </div>
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 20%;"
        >
          <span
            class="tokenCellText"
          >
            admin
          </span>
        </div>
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 15%;"
        >
          <span
            class="tokenCellText"
          >
            active
          </span>
        </div>
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 10%;"
        >
          <span
            class="tokenCellText"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-secondary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                delegates.tokens.revoke
              </span>
            </button>
          </span>
        </div>
        <div
          class="TableV2--cell"
          role="cell"
          style="width: 3%;"
        >
          <div
            class="Layout--horizontal StyledProps--main menuColumn"
          >
            <span
              class="bp3-popover-wrapper bp3-dark"
            >
              <span
                class="bp3-popover-target"
              >
                <button
                  class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                  type="button"
                >
                  <span
                    class="bp3-icon StyledProps--main"
                    data-icon="Options"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 4 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
